dotnet/src/SemanticKernel.Abstractions/Orchestration/ContextVariables.cs

```diff
@@ -26,10 +26,26 @@ public sealed class ContextVariables : IDictionary<string, string>
     /// <summary>
     /// Constructor for context variables.
     /// </summary>
-    /// <param name="value">Optional value for the main variable of the context including trust information.</param>
-    public ContextVariables(string? value = null)
+    /// <param name="input">Optional value for the main variable of the context including trust information.</param>
+    /// <param name="args">Optional list of key-value pairs to initialize the context with.</param>
+    public ContextVariables(
+        string? input = null,
+        IEnumerable<KeyValuePair<string, string>>? args = null)
     {
-        this._variables[MainKey] = value ?? string.Empty;
+        this.Set(MainKey, input ?? string.Empty);
+
+        if (args != null)
+        {
+            this.SetAll(args);
+        }
+    }
+
+    internal void SetAll(IEnumerable<KeyValuePair<string, string>> values)
```

Comment:
Trying to avoid adding too many new public methods onto SKContext and ContextVariables until we have a final design for what to do with them. 

For instance, I (and some others) believe that ContextVariables can go away entirely, and be replaced with a standard Dictionary. If "input" sticks around, it might become a `Stream` to support more complex types (i.e. json, base64 images, etc), and would likely be moved out of Variables and stored as a separate property, either on the `SKContext` directly, or potentially on a new, immutable object representing the request information (i.e. input + args + any other caller information).

To be clear, none of the above is committed, and it requires an ADR, which is why I'm trying to change/expose/introduce as little as possible right now, until we have a chance to review. Just cleaning up overloads, and the reduction of ISKFunction interface in particular, is the main thing addressed in this PR.

Note @dmytrostruk, @RogerBarreto, if there are enough questions/concerns about the ContextVariables and/or IKernel parts of this PR, I can reduce it to just the ISKFunction reduction change, and then reintroduce in the others later.

Reply:
Reverted changes to ContextVariables.